daemon/containerd/resolver.go

```diff
@@ -1,35 +1,89 @@
 package containerd
 
 import (
+	"net/http"
+	"strings"
+
 	"github.com/containerd/containerd/remotes"
 	"github.com/containerd/containerd/remotes/docker"
 	registrytypes "github.com/docker/docker/api/types/registry"
 	"github.com/docker/docker/registry"
 	"github.com/sirupsen/logrus"
 )
 
-func newResolverFromAuthConfig(authConfig *registrytypes.AuthConfig) remotes.Resolver {
-	opts := []docker.RegistryOpt{}
-	if authConfig != nil {
-		cfgHost := registry.ConvertToHostname(authConfig.ServerAddress)
-		if cfgHost == registry.IndexHostname {
-			cfgHost = registry.DefaultRegistryHost
+func (i *ImageService) newResolverFromAuthConfig(authConfig *registrytypes.AuthConfig) (remotes.Resolver, docker.StatusTracker) {
+	tracker := docker.NewInMemoryTracker()
```

Comment:
The tracker is optional. If a caller doesn't care about reusing the tracker, [they can pass `nil`.](https://github.com/containerd/containerd/blob/5b842e528e99d4d4c1686467debf2bd4b88ecd86/remotes/docker/resolver.go#L136-L139)
```suggestion
func (i *ImageService) newResolverFromAuthConfig(authConfig *registrytypes.AuthConfig, tracker docker.StatusTracker) remotes.Resolver {
```

Reply:
The shared tracker will be needed in the near future when I upstream [image pushing](https://github.com/rumpl/moby/blob/c8d/daemon/containerd/image_push.go#L63)